File: ios/Classes/Data/LocationOptions.m
//
//  LocationOptions.m
//  geolocator
//
//  Created by Maurits van Beusekom on 20/05/2019.
//

#import "LocationOptions.h"

@implementation LocationOptions
- (instancetype)init {
    self = [super init];
    if (self) {
        self.accuracy = GeolocationAccuracyBest;
        self.distanceFilter = 0;
    }
    
    return self;
}


- (instancetype)initWithAccuracy:(GeolocationAccuracy)accuracy distanceFilter:(CLLocationDistance)distanceFilter {
    self = [super init];
    if (self) {
        self.accuracy = accuracy;
        self.distanceFilter = distanceFilter;
    }
    
    return self;
}

- (instancetype)initWithArguments:(id)arguments {
    self = [super init];
    if (self) {
        if ([arguments isKindOfClass:[NSDictionary class]]) {
            NSNumber *accuracy = arguments[@"accuracy"];
            NSNumber *distanceFilter = arguments[@"distanceFilter"];
            
            self.accuracy = (GeolocationAccuracy) accuracy.intValue;
            self.distanceFilter = distanceFilter.doubleValue;
        }
    }
    
    return self;
}

@end

File: ios/Classes/Data/LocationOptions.swift
//
//  LocationOptions.swift
//  geolocator
//
//  Created by Maurits van Beusekom on 11/07/2018.
//

import Foundation
import CoreLocation

struct LocationOptions : Codable {
    init() {
        self.accuracy = GeolocationAccuracy.best
        self.distanceFilter = 0
    }
    
    var accuracy: GeolocationAccuracy
    var distanceFilter: CLLocationDistance
}

enum GeolocationAccuracy : String, Codable {
    case lowest = "lowest"
    case low = "low"
    case medium = "medium"
    case high = "high"
    case best = "best"
    case bestForNavigation = "bestForNavigation"
    
    var clValue: CLLocationAccuracy {
        switch self {
        case .lowest:
            return kCLLocationAccuracyThreeKilometers
        case .low:
            return kCLLocationAccuracyKilometer
        case .medium:
            return kCLLocationAccuracyHundredMeters
        case .high:
            return kCLLocationAccuracyNearestTenMeters
        case .best:
            return kCLLocationAccuracyBest
        case .bestForNavigation:
            return kCLLocationAccuracyBestForNavigation
        }
    }
}
